airflow/providers/qubole/operators/qubole_check.py

```diff
@@ -16,17 +16,40 @@
 # specific language governing permissions and limitations
 # under the License.
 #
-from typing import Iterable, Optional, Union
+from typing import Callable, Iterable, Optional, Union
 
 from airflow.exceptions import AirflowException
-from airflow.operators.check_operator import CheckOperator, ValueCheckOperator
-from airflow.providers.qubole.hooks.qubole import QuboleHook
+from airflow.operators.sql import SQLCheckOperator, SQLValueCheckOperator
 from airflow.providers.qubole.hooks.qubole_check import QuboleCheckHook
 from airflow.providers.qubole.operators.qubole import QuboleOperator
 from airflow.utils.decorators import apply_defaults
 
 
-class QuboleCheckOperator(CheckOperator, QuboleOperator):
+class _QuboleCheckOperatorMixin:
+    """This is a Mixin for Qubole related check operators"""
+
+    def execute(self, context=None) -> None:
+        """Execute a check operation againt Qubole"""
+        try:
+            self._hook_context = context
+            super().execute(context=context)
+        except AirflowException as e:
+            handle_airflow_exception(e, self.get_hook())
+
+    def get_db_hook(self) -> QuboleCheckHook:
+        """Get QuboleCheckHook"""
+        return self.get_hook()
+
+    # this overwrite the original QuboleOperator.get_hook() which returns a QuboleHook.
+    def get_hook(self) -> QuboleCheckHook:
+        """Reinitialising the hook, as some template fields might have changed"""
+        return QuboleCheckHook(
+            context=self._hook_context, results_parser_callable=self.results_parser_callable, **self.kwargs
+        )
+
+
+# pylint: disable=too-many-ancestors
```

Comment:
Temporarily disable pylint for these 2 operators. Will submit a second PR in the future to simplify the inheritance.

Reply:
Yeah. for now it is good, but maybe leave a TODO(xinbinhuang)  note as comment?